<launch>

    <group>
        <remap from="generic_setpoint" to="ang_setpoint"/>
        <remap from="current_state" to="ang_state"/>
        <remap from="pid_output" to="ang_pid_output"/>
        
        <node pkg="lab1" name="anglePID" type="PID.py">
            <remap from="p_term" to="ang/p_term"/>
            <remap from="i_term" to="ang/i_term"/>
            <remap from="d_term" to="ang/d_term"/>
        </node>
        
        <param name="ang/p_term" type="double" value="1.5"/>
        <param name="ang/i_term" type="double" value="0.05"/>
        <param name="ang/d_term" type="double" value="3.0"/>
    </group>
    
     <group>
        <remap from="generic_setpoint" to="dist_setpoint"/>
        <remap from="current_state" to="dist_state"/>
        <remap from="pid_output" to="dist_pid_output"/>
        
        <node pkg="lab1" name="distancePID" type="PID.py">
            <remap from="p_term" to="dist/p_term"/>
            <remap from="i_term" to="dist/i_term"/>
            <remap from="d_term" to="dist/d_term"/>
        </node>
        
        <param name="dist/p_term" type="double" value="3.0"/>
        <param name="dist/i_term" type="double" value="0.05"/>
        <param name="dist/d_term" type="double" value="1.72"/>
    </group>

    <param name="linear_input_scaling" type="double" value="1.0"/>
    <param name="angular_input_scaling" type="double" value="1.0"/>

    <node pkg="balboa_core" name="balboa_serial" type="balboa_serial.py" />
    <node pkg="lab2" name="drive_and_sample" type="drive_and_sample.py" output="screen" />
    <node pkg="lab1" name="input_capture" type="input_capture.py"/>
    <node pkg="lab1" name="convert_sensor_input" type="convert_sensor_input.py"/>
    <node pkg="lab1" name="adder" type="adder.py"/>

</launch>
